components/infobox/extensions/commons/infobox_extensions_prize_pool.lua

```diff
@@ -0,0 +1,161 @@
+			prizepool = prizepool,
+			prizepoolUsd = prizepoolUsd,
+		}
+		if errorMessage then
+			return PrizePoolCurrency._errorMessage(errorMessage)
+		end
+	end
+
+	if setVariables then
+		if String.isNotEmpty(args.currency) and currencyRate then
+			Variables.varDefine(currency .. '_rate', currencyRate)
+		end
+		if String.isNotEmpty(args.currency) then
+			Variables.varDefine('tournament_currency', currency:upper())
+		end
+
+		Variables.varDefine('tournament_currency_date', date)
+		Variables.varDefine('tournament_currency_text', text)
+		Variables.varDefine('tournament_prizepoollocal', prizepool)
+
+		local prizepoolUsdValue = string.gsub(prizepoolUsd or '', ',', '')
+		Variables.varDefine('tournament_prizepoolusd', prizepoolUsdValue)
+
+		-- legacy compatibility
+		Variables.varDefine('tournament_currency_rate', currencyRate or '')
+		Variables.varDefine('tournament_prizepool_local', prizepool)
+		Variables.varDefine('tournament_prizepool_usd', prizepoolUsd)
+		Variables.varDefine('currency', args.currency and currency)
+		Variables.varDefine('currency date', date)
+		Variables.varDefine('currency rate', currencyRate)
+		Variables.varDefine('prizepool', prizepool)
+		Variables.varDefine('prizepool usd', prizepoolUsd)
+		Variables.varDefine('tournament_prizepool', prizepoolUsdValue)
+	end
+
+	local display = Currency.display(USD, prizepoolUsd, {formatValue = false, setVariables = false})
+	if String.isNotEmpty(prizepool) then
+		display = Currency.display(currency, prizepool, {formatValue = false, setVariables = true})
+			.. '<br>(≃ ' .. display .. ')'
+	end
+
+	return display
+end
+
+function PrizePoolCurrency._exchange(props)
+	local prizepool = props.prizepool or ''
+	local prizepoolUsd = props.prizepoolUsd or ''
+	local currencyRate = Currency.getExchangeRate(props)
+
+	if not currencyRate then
+		local errorMessage = 'Need valid exchange date'
+		return errorMessage, prizepool, prizepoolUsd, currencyRate
+	end
+
+	if currencyRate and Logic.isNumeric(prizepool) and currencyRate ~= math.huge then
+		prizepoolUsd = tonumber(prizepool) * currencyRate
+	end
+
+	if
+		not currencyRate
+		and not Logic.isNumeric(prizepool)
+		and not Logic.isNumeric(prizepoolUsd)
+	then
+		local errorMessage = 'Need valid currency, exchange rate'
+		return errorMessage, prizepool, prizepoolUsd, currencyRate
+	end
+
+	if Logic.isNumeric(prizepool) then
+		prizepool = PrizePoolCurrency._format(prizepool)
+	end
+	if Logic.isNumeric(prizepoolUsd) then
+		prizepoolUsd = PrizePoolCurrency._format(prizepoolUsd)
+	end
+
+	return nil, prizepool, prizepoolUsd, currencyRate
+end
+
+function PrizePoolCurrency._format(value)
+	return LANG:formatNum(math.floor(value))
+end
+
+function PrizePoolCurrency._cleanDate(dateString)
+	dateString = string.gsub(dateString, '[^%d.-]', '')
+
+	return string.match(dateString, '%d%d%d%d%-%d%d%-%d%d')
+end
+
+function PrizePoolCurrency._cleanValue(valueString)
+	return tonumber(string.gsub(valueString or '', '[^%d.?]', '') or '')
```

Comment:
Does this actually work, considering the `.` in there?

Reply:
tbh it is a plain copy paste from the live one, will check

Reply:
i think they wanted `%.` (to allow 100.50)